generator/php/resources/templates/ClientCredentialsClient.mustache

```diff
@@ -153,18 +157,121 @@ class ClientCredentialsClient implements \GuzzleHttp\ClientInterface
     {
         if ($this->token == null || !$this->token->isValidEnough()) {
             try {
-                $response = $this->oauthApi->getToken(GRANT_TYPE, $this->clientId, $this->clientSecret);
+                $response = $this->oauthClient->getToken(GRANT_TYPE, $this->clientId, $this->clientSecret);
             } catch (ApiException $e) {
                 throw new ApiException('Cannot refresh token automatically.', $e->getCode(), $e->getResponseHeaders(), $e->getResponseBody());
             }
-            $this->token = new Token($response->getAccessToken(), $response->getExpiresIn());
+            $this->token = new Token($response->access_token, $response->expires_in);
         }
         return $request
             ->withoutHeader(AUTHORIZATION)
             ->withAddedHeader(AUTHORIZATION, 'Bearer ' . $this->token->getValue());
     }
 }
 
+class OAuthClient{
+    /**
+     * @var ClientInterface
+     */
+    protected $client;
+
+    /**
+     * @var Configuration
+     */
+    protected $config;
+
+    /**
+     * OAuthClient constructor.
+     * @param ClientInterface|null $client
+     * @param Configuration|null $config
+     */
+    public function __construct(
+        $client = null, 
+        $config = null)
+    {
+        $this->client = $client ?: new Client();
+        $this->config = $config ?: new Configuration();
+    }
+
+    public function getToken($grant_type, $client_id, $client_secret, )
+    {
+        list($response) = $this->getTokenWithHttpInfo($grant_type, $client_id, $client_secret);
+        return $response;
+    }
+
+    /**
+     * Creates a token based on supplied client credentials
+     *
+     * @param  string $grant_type (required)
+     * @param  string $client_id (required)
+     * @param  string $client_secret (required)
+     *
+     * @return array
+     */
+    private function getTokenWithHttpInfo($grant_type, $client_id, $client_secret)
```

Comment:
OPT: this method is used only once (on L198 just above) in a method which just calls it, and it is private. Perhaps it could be inlined (for clarity)?